crates/uv/src/commands/project/init.rs

```diff
@@ -92,6 +102,90 @@ pub(crate) async fn init(
         }
     };
 
+    // Add a `requires-python` field to the `pyproject.toml`.
+    let requires_python = if let Some(request) = python.as_deref() {
+        // (1) Explicit request from user
+        match PythonRequest::parse(request) {
+            PythonRequest::Version(VersionRequest::MajorMinor(major, minor)) => {
+                RequiresPython::greater_than_equal_version(&Version::new([
+                    major as u64,
+                    minor as u64,
+                ]))
+            }
+            PythonRequest::Version(VersionRequest::MajorMinorPatch(major, minor, patch)) => {
+                RequiresPython::greater_than_equal_version(&Version::new([
+                    major as u64,
+                    minor as u64,
+                    patch as u64,
+                ]))
+            }
+            PythonRequest::Version(VersionRequest::Range(specifiers)) => {
+                RequiresPython::from_specifiers(&specifiers)?
+            }
+            request => {
+                let reporter = PythonDownloadReporter::single(printer);
+                let client_builder = BaseClientBuilder::new()
+                    .connectivity(connectivity)
+                    .native_tls(native_tls);
+                let interpreter = PythonInstallation::find_or_fetch(
+                    Some(request),
+                    EnvironmentPreference::Any,
+                    python_preference,
+                    python_fetch,
+                    &client_builder,
+                    cache,
+                    Some(&reporter),
+                )
+                .await?
+                .into_interpreter();
+                RequiresPython::greater_than_equal_version(&interpreter.python_minor_version())
```

Comment:
@zanieb - Can you review that what I'm doing with the discovery here is reasonable?